llvm/lib/Target/AMDGPU/AMDGPULowerBufferFatPointers.cpp

```diff
@@ -1804,14 +1804,26 @@ PtrParts SplitPtrStructs::visitGetElementPtrInst(GetElementPtrInst &GEP) {
   bool IsNUW = GEP.hasNoUnsignedWrap();
   bool IsNUSW = GEP.hasNoUnsignedSignedWrap();
 
+  Type *ResTy = GEP.getType();
+  std::optional<ElementCount> ResEC;
```

Comment:
I don't see the point of ResEC, can just use the `dyn_cast<VectorType>` result later?

Reply:
Should be fixed